src/components/HbLoadingSpinner.tsx

```diff
@@ -0,0 +1,90 @@
+import random from "lodash/random";
+import React, { ReactNode, useContext, useMemo } from "react";
+import { Css } from "src/Css";
+import SpinnerGifBase64 from "./HbLoadingSpinner.base64";
+
+interface HbLoadingSpinnerProps {
+  /** Reverts loading text to `Loading...` if true. May override global noQuips by passing in noQuips={false}. */
+  noQuips?: boolean;
+  /** Contextually relevant quips, like "Where did I put that invoice??" on some invoice page. Will override global noQuips if non-empty. */
+  extraQuips?: ReadonlyArray<string>;
+  extraQuipsOnly?: boolean;
+  /** No text at all */
+  iconOnly?: boolean;
+}
+
+export function HbLoadingSpinner({ noQuips, extraQuips = [], extraQuipsOnly, iconOnly }: HbLoadingSpinnerProps) {
+  const ctx = useContext(HbLoadingSpinnerContext);
+
+  const quip = useMemo(() => {
+    // If quips are off globally but custom quips were provided, then override global
+    const forceQuips = extraQuips.length !== 0 || noQuips === false;
+    if ((ctx.noQuips && !forceQuips) || noQuips) {
+      return "Loading...";
+    }
+
+    const allQuips = extraQuipsOnly && extraQuips.length !== 0 ? extraQuips : [...ctx.quips, ...extraQuips];
+    return allQuips[random(allQuips.length - 1)];
```

Comment:
**suggestion:** Could we remove the need of `lodash` as a dependency and use `Math.random` here instead?
```suggestion
    return allQuips[Math.floor(Math.random() * allQuips.length)];
```

Reply:
Vite tree-shakes it out when you import modules using the `import default from 'lodash/specific-module'` format.

Reply:
^ For which we have an ESLint rule

Reply:
Yea, it is good we can tree shake out unnecessary modules, but my push for this is that we don't need an extra dependency to accomplish the same thing, especially if it the same number of lines of code.
Out of curiosity I took a look at `lodash`'s [random](https://github.com/lodash/lodash/blob/master/random.js) script. Definitely more than we need, though still quite small. Either way, if using native javascript to accomplish the same task is as many lines of code, then I'd push to use the native solution.